CMakeLists.txt

```diff
@@ -24,7 +24,7 @@ macro(build_libyaml)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
   endif()
 
-  list(APPEND extra_cmake_args -DBUILD_SHARED_LIBS=ON)
+  list(APPEND extra_cmake_args -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS})
```

Comment:
@allsey87 nit: consider forwarding `BUILD_SHARED_LIBS` only if defined. That way, we get to keep default behavior as defined upstream.

Reply:
This can't be right as-is, can it?  What defines `BUILD_SHARED_LIBS` in the default case?

Reply:
> This can't be right as-is, can it? What defines BUILD_SHARED_LIBS in the default case?

Not defined `BUILD_SHARED_LIBS` is equivalent to false (see [here](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html)).

Reply:
Shouldn't `BUILD_SHARED_LIBS` always be defined? https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Reply:
We can change it to this then:
```cmake
if(BUILD_SHARED_LIBS)
  list(APPEND extra_cmake_args -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS})
else()
  list(APPEND extra_cmake_args)
endif()
```
The question is then what to do in the `else` block. Should `BUILD_SHARED_LIBS` remain hard coded to `ON` to match the current behavior?

Reply:
> Should BUILD_SHARED_LIBS remain hard coded to ON to match the current behavior?

I would not set it at all and leave that to upstream `libyaml`, but considering it used to be forced to `ON` let's keep it that way.